.github/actions/get-test-plan-key/action.yml

```diff
@@ -0,0 +1,75 @@
+name: get-test-plan-key
+description: "Determine the TP key if the TP exist, create it if doesn't"
+inputs:
+  major_version:
+    required: true
+    type: string
+  xray_token:
+    required: true
+    type: string
+outputs:
+  test_plan_key:
+    description: "the test plan key"
+    value: ${{ steps.get_test_plan_key.outputs.test_plan_key }}
+
+# NIGHTLY <OSS|MODULES|COLLECT> <MAJOR_VERSION> (e.g., NIGHTLY OSS 23.10)
+runs:
+  using: "composite"
+  steps:
+    - name: Get the TestPlanKey
+      id: get_test_plan_key
+      run: |
+        # Use a GraphQL query to get all test plans matching the JQL
+        graphql_query='{
+          "query":"query GetTestPlans($jql: String, $limit: Int!) { getTestPlans(jql: $jql, limit: $limit) { total results { issueId jira(fields: [\"summary\", \"key\"]) } } }",
+          "variables":{"jql": "project = MON","limit": 100}}'
```

Comment:
```suggestion
          "variables":{"jql": "project = MON","limit": $limit}}'
```

Reply:
@sc979, it won't work because `$limit` is a variable that hasn't been set.

Reply:
Ah indeed I read it wrong